docs/book/src/cronjob-tutorial/testdata/project/Makefile

```diff
@@ -52,16 +52,8 @@ manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and Cust
 generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
 	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
 
-.PHONY: fmt
-fmt: ## Run go fmt against code.
-	go fmt ./...
```

Comment:
are you sure with that? I think it was good practice call formatting for every build/test/run. Lint is optional and fix too. So you cannot forget call this targets before pushing changes otherwise you can create many changes in commits.

I personally suggest keep fmt and don't have dependency to third party software.
it can help avoid unnecessary changes in PR.

Vet can be deleted.

Reply:
Additional complication can be linter config. Linter config is manage by user and he can very easy disable fmt linter. This hardcoded way in makefile looks better for me.

Reply:
> I personally suggest keep fmt and don't have dependency to third party software.
it can help avoid unnecessary changes in PR.

Agreed, will add back!

I was thinking about directly calling `go fmt` inside `make generate`, and still remove `fmt` command from Makefile (without changing `make generate` current behavior). This way, kubebuilder CI would still prevent unnecessary changes in PRs, and users would have more control over generated code. 

What you think?

Reply:
i am not sure. you will create dependency between formatting and generating what are independent process (they depend on the code).

Is it okay, if fmt fail and generate won't run but independently it can run?
What can be weird is make target  name. It is generate so i expect it will create something but fmt just modify current state.

I am not sure and i think we are doing overengineering. Did you heard some complain about current solution?
I personally don't see big benefits in vet because it is linter and it is part of golangci-lint so it can be deleted.
But other stuff i kept as it is because i don't see big benefits and it can create just new problems.

I think opinion from someelse can be helpful.

Reply:
My only complain is that  `fmt` command does not have a CI counterpart, so I created one in the past (called it `fmt-isuptodate`) in my project Makefile. After understanding `golangci-lint`, I removed `fmt` and `fmt-isuptodate` and started using `lint-fix` and `lint` respectively, in the project Makefile.

> What can be weird is make target name. It is generate so i expect it will create something but fmt just modify current state. 

hmmm, one other view we can have then, is updating generate command to generate the code already formatted? Looking in the generated Makefile command, the two code generation commands, `generate` and `manifests` do not run `fmt` (but I think Go is smart enough to ignore format in generated files by controller-gen, so this can be bad example :grimacing: )

Reply:
CI counterpart. CI can be extremely different and it depends from environment and CI/CD software. I am using gitlab and i have own ci which just call make fmt in projects, so i never had problem with ci/cd. It exists a lot of CI/CD solutions and i don't think it is good idea force some ci/cd solution into this project, which is technically template for another projects. I think our main target has to be easy usage for local development.

Reply:
alright! I think today I will not have time, but tomorrow I will commit changes based on our conversation :grinning:

Reply:
@lukas016 just to keep you posted: I found some problems in scripts that generate docs testdata projects. I am working on fixing them. Once I finished (this week, but not sure which day), I will open a PR for that and update this one to just remove `vet` command (reverting removal of `fmt` command)